### ISSUE TITLE
Missing method SingletonAttribute.get_Scope() putting messages into poison queue

### ISSUE THREAD

**Issue by markjbrown** [Azure/azure-webjobs-sdk#636]
I have a continuously running WebJob (v 1.1.0) that listens to a queue for new messages. It is part of a larger solution ([AzureDmlBackup](https://github.com/markjbrown/AzureDmlBackup)) with one on-demand and another cron scheduled WebJob.

I had some issues last week with nuget packages for this solution but I thought I had resolved them. I have deleted and restored the packages directory.

I am getting the below text in the Output log for the job. The WebJob no longer processes any queue messages. All queue messages are now dumped to the poison queue.

I took the time stamps out to make the log look a bit less wordy.

```
System.MissingMethodException: Method not found: 'System.String Microsoft.Azure.WebJobs.SingletonAttribute.get_Scope()'.
at Microsoft.Azure.WebJobs.Host.Triggers.TriggeredFunctionBinding`1.<BindCoreAsync>d__7.MoveNext()
at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start[TStateMachine](TStateMachine& stateMachine)
at Microsoft.Azure.WebJobs.Host.Triggers.TriggeredFunctionBinding`1.BindCoreAsync(ValueBindingContext context, Object value, IDictionary`2 parameters)
at Microsoft.Azure.WebJobs.Host.Triggers.TriggeredFunctionBinding`1.<BindAsync>d__0.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.<ExecuteWithLogMessageAsync>d__c.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.<TryExecuteAsync>d__1.MoveNext()
Function had errors. See Azure WebJobs SDK dashboard for details. Instance ID is 'd5d080b5-c595-4ac5-a0da-72f3192e3791'
```

**Comment by davidebbo** [comment]
Strange thing is that the Scope attribute was added in v1.1.1 (This commit: 56a5d491ecdd50ffdd41a83b485f94334c6e093e). So it seems even though you're running v1.1.0, there is something that has v1.1.1 expectations.

**Comment by markjbrown** [comment]
If I do a redeploy in App Services does it remove any former or unused Nuget packages?

I had formerly had 1.1.1 in the two on demand web jobs but now have everything at 1.1.0

Sent from Outlook Mobile<https://aka.ms/qtex0l>




On Mon, Feb 22, 2016 at 11:00 AM -0800, "David Ebbo" <notifications@github.com<mailto:notifications@github.com>> wrote:


Strange thing is that the Scope attribute was added in v1.1.1 (This commit: 56a5d49<https://github.com/Azure/azure-webjobs-sdk/commit/56a5d491ecdd50ffdd41a83b485f94334c6e093e>). So it seems even though you're running v1.1.0, there is something that has v1.1.1 expectations.

-

**Comment by davidebbo** [comment]
It should, though that may depend how you deploy. See [this page](https://github.com/projectkudu/kudu/wiki/Deployment-vs-runtime-issues) for investigation tips.

**Comment by markjbrown** [comment]
That link was helpful but I'm still stuck. I downloaded wwwroot and I found that version 1.1.1 was indeed deployed into the web host which is incorrect. It should be 1.1.0 since I rolled everything back.

To fix this I deleted the WebJobs from the Portal which removed them from the deployment. I then redeployed them from VS with version 1.1.0 but I am still seeing version 1.1.1 deployed in App Services. When I deploy this I deploy the the entire solution by right clicking and selecting Publish for host website in Visual Studio. This deploys any updated Web Jobs with it.

However I am still seeing version 1.1.1 deployed. 

Any thoughts?

**Comment by markjbrown** [comment]
So I think I see the problem but I am unsure how to fix it and who to complain to.

When I look in packages.config I see the correct version (1.1.0) for WebJobs.

`<package id="Microsoft.Azure.WebJobs" version="1.1.0" targetFramework="net452" />`
`<package id="Microsoft.Azure.WebJobs.Core" version="1.1.0" targetFramework="net452" />`

However, when I look in References in VS I see 1.1.1.

So why is VS not properly updating the references in the csproj file?

![image](https://cloud.githubusercontent.com/assets/800166/13235666/01ffa6be-d974-11e5-9cf3-28e20b4edecd.png)

**Comment by markjbrown** [comment]
I attempted to fix this issue by uninstalling and reinstalling all the packages I am encountering another error.

Looks like I borked this up pretty good :)

```
[02/23/2016 00:43:26 > bfd756: SYS INFO] Run script 'DmlExec.exe' with script host - 'WindowsScriptHost'
[02/23/2016 00:43:26 > bfd756: SYS INFO] Status changed to Running
[02/23/2016 00:43:27 > bfd756: ERR ] 
[02/23/2016 00:43:27 > bfd756: ERR ] Unhandled Exception: System.TypeInitializationException: The type initializer for 'Microsoft.Azure.WebJobs.Host.Indexers.DefaultTypeLocator' threw an exception. ---> System.IO.FileLoadException: Could not load file or assembly 'Microsoft.Azure.WebJobs, Version=1.1.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040) ---> System.IO.FileLoadException: Could not load file or assembly 'Microsoft.Azure.WebJobs, Version=1.1.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)
[02/23/2016 00:43:27 > bfd756: ERR ]    --- End of inner exception stack trace ---
```

**Comment by davidebbo** [comment]
Seems it's still trying to use 1.1.1 (`Could not load file or assembly 'Microsoft.Azure.WebJobs, Version=1.1.1`). I'm not sure what's going on, but it seems like more of a VS/NuGet issue than anything directly related to the SDK.

**Comment by markjbrown** [comment]
I don't know what could be the issue with VS or nuget. I have looked at everything and it is all using Web Jobs 1.1.0. My source is on github if you want to take a look. https://github.com/markjbrown/AzureDmlBackup

**Comment by davidebbo** [comment]
Couple things to try to isolate:

- Does it repro if you deploy to a new test site? If so, that would mean it's not related to a problem on the WebApp
- Does it repro if you do a 'deploy to local file system' in VS?

**Comment by markjbrown** [comment]
I tested this in a new web host with the same results.

I then went through the entire app again and found that app.config had been modified to include a new  <dependentAssembly> section that was forcing the app to use 1.1.1 again. Feels like playing "Whack-a-Mole" with Nuget and VS sometimes.

I removed this and redeployed and the application is working normally again.

Thanks for helping to track down. I apologize this was so difficult. I am looking forward to the updated WebJobs SDK to remove the Storage Library version celing.

Thanks David
